### PR TITLE
Added NL prompt 'manual mode' to require user to hit enter before showing the next query

### PR DESCRIPTION
**Pull request by dwnoble**
* Manual mode is enabled with a new URL paramter `m=True` or `m=1`
* Added "query delay" (`qd`) parameter to configure the amount of time to wait before the next query to display in non-manual mode. Default: 2000ms. Example, set the query delay to 5s: `qd=5000`
* Refactored code to avoid direct dom manipulations in a few places

Tested with:

go/dcnl-main-demo
go/dcnl-sdg-demo
go/dcnl-me-demo
go/dcnl-ct-demo

### CODE REVIEW

File: static/js/apps/nl_interface/query_context.tsx [hunk 1]
```diff
@@ -59,41 +69,96 @@ export function QueryContext(): JSX.Element {
...
+    ]
+  );
+  function inputNextPrompt2(delayStart: boolean): void {
+    const prompt = urlPrompts[autoPlayCurrentQueryIndex];
+    console.log(
+      "In inputNextPrompt... autoPlayCurrentQueryIndex=",
+      autoPlayCurrentQueryIndex,
+      "prompt=",
+      prompt
+    );
```
Comment (pradh) [inline static/js/apps/nl_interface/query_context.tsx:130]: nit: drop

File: static/js/apps/nl_interface/query_context.tsx [hunk 2]
```diff
@@ -59,41 +69,96 @@ export function QueryContext(): JSX.Element {
...
+      autoPlayAutoShowQueryDelay,
+      autoPlayAutoRunQuery,
+      currentNlQueryContext,
+      nlQueryHistory,
+      urlPrompts,
+      search,
+      setAutoRunQuery,
+    ]
+  );
+  function inputNextPrompt2(delayStart: boolean): void {
```
Comment (pradh) [inline static/js/apps/nl_interface/query_context.tsx:123]: maybe func-level comment on what this does?

File: static/js/apps/nl_interface/query_context.tsx [hunk 3]
```diff
@@ -59,41 +69,96 @@ export function QueryContext(): JSX.Element {
...
+    const promptSearchDelay = autoPlayDisableTypingAnimation
+      ? 0
+      : PROMPT_SEARCH_DELAY;
     const nextPromptDelay =
-      delayStart && !delayDisabled ? NEXT_PROMPT_DELAY : 0;
-    const promptSearchDelay = delayDisabled ? 0 : PROMPT_SEARCH_DELAY;
+      delayStart && !autoPlayDisableTypingAnimation
+        ? autoPlayAutoShowQueryDelay
+        : 0;
+    console.log("nextPromptDelay=", autoPlayAutoShowQueryDelay);
```
Comment (pradh) [inline static/js/apps/nl_interface/query_context.tsx:140]: drop, and below...

**Review by pradh — APPROVED** [review]
Thanks!

File: static/js/apps/nl_interface/query_context.tsx (hunk 2 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/query_context.tsx]: Whoops- meant to drop this function all together. Dropped!

File: static/js/apps/nl_interface/app_state.ts [hunk 4]
```diff
@@ -75,15 +75,58 @@ export interface NLQueryContextType {
...
+  autoPlayAutoRunQuery: boolean;
+
+  /**
+   * Auto play mode option: Wait for the user to hit "enter" in the search box before showing the next query
+   * Enable with URL param: "m=True"
+   * Default: false
+   */
+  autoPlayManuallyShowQuery: boolean;
+
+  /**
```
Comment (shifucun) [inline static/js/apps/nl_interface/app_state.ts:92]: &m and &qd are mutually exclusive? Wonder should &m has higher priority.

File: static/js/apps/nl_interface/app_state.ts [hunk 5]
```diff
@@ -75,15 +75,58 @@ export interface NLQueryContextType {
...
+   * Auto play mode internal state: current query index
+   */
+  autoPlayCurrentQueryIndex: number;
+
+  /**
+   * Auto play mode option: Disable the typing animation on auto-play  mode
+   * Enable with URL param "d=True"
+   * Default: false
+   */
+  autoPlayDisableTypingAnimation: boolean;
```
Comment (shifucun) [inline static/js/apps/nl_interface/app_state.ts:109]: maybe name it autoPlayEnableTypingAnimation for easy understanding?

File: static/js/apps/nl_interface/app_state.ts [hunk 6]
```diff
@@ -75,15 +75,58 @@ export interface NLQueryContextType {
...
   currentNlQueryContextId: string;
-  delayDisabled: boolean;
-  demoMode: boolean; // Hide feedback buttons
+
+  /**
+   * Hides feedback buttons.
+   * Enable with URL param "enable_demo=True"
+   * Default: false
+   */
+  hideFeedbackButtons: boolean;
```
Comment (shifucun) [inline static/js/apps/nl_interface/app_state.ts:118]: same here.. showFeedbackButtons

File: static/js/apps/nl_interface/query_context.tsx [hunk 7]
```diff
@@ -59,67 +69,91 @@ export function QueryContext(): JSX.Element {
...
-          // If on autorun, search for the current input after
-          // PROMPT_SEARCH_DELAY ms.
-          if (autoRun) {
-            searchDelayTimer.current = setTimeout(() => {
-              executeSearch();
-            }, promptSearchDelay);
+  const inputNextPrompt = useCallback(
+    (delayStart: boolean): void => {
+      const prompt = urlPrompts[autoPlayCurrentQueryIndex];
+      //const nextPromptDelay = 0;
```
Comment (shifucun) [inline static/js/apps/nl_interface/query_context.tsx:75]: remove comment?

**Review by shifucun — APPROVED** [review]
nice!

File: static/js/apps/nl_interface/query_context.tsx (hunk 7 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/query_context.tsx:75]: done!

File: static/js/apps/nl_interface/app_state.ts (hunk 6 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/app_state.ts:118]: leaving as is per offline conversation

File: static/js/apps/nl_interface/app_state.ts (hunk 5 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/app_state.ts:109]: leaving as is per offline conversation

File: static/js/apps/nl_interface/app_state.ts (hunk 4 above)
Comment (dwnoble) [reply static/js/apps/nl_interface/app_state.ts:92]: Agreed about m's precedent - added!

**Pull request merged**